buildSrc/src/main/java/org/opensearch/gradle/docker/DockerSupportService.java

```diff
@@ -98,48 +103,61 @@ public DockerSupportService(ExecOperations execOperations) {
      */
     public DockerAvailability getDockerAvailability() {
         if (this.dockerAvailability == null) {
-            String dockerPath = null;
-            Result lastResult = null;
-            Version version = null;
-            boolean isVersionHighEnough = false;
-            boolean isComposeAvailable = false;
-
-            // Check if the Docker binary exists
-            final Optional<String> dockerBinary = getDockerPath();
-            if (isExcludedOs() == false && dockerBinary.isPresent()) {
-                dockerPath = dockerBinary.get();
-
-                // Since we use a multi-stage Docker build, check the Docker version meets minimum requirement
-                lastResult = runCommand(dockerPath, "version", "--format", "{{.Server.Version}}");
-
-                if (lastResult.isSuccess()) {
-                    version = Version.fromString(lastResult.stdout.trim(), Version.Mode.RELAXED);
-
-                    isVersionHighEnough = version.onOrAfter(MINIMUM_DOCKER_VERSION);
-
-                    if (isVersionHighEnough) {
-                        // Check that we can execute a privileged command
-                        lastResult = runCommand(dockerPath, "images");
+            // Gradle BuildService implementations must be thread-safe
+            synchronized (this) {
```

Comment:
This is interesting, I think we discussed that @lukas-vlcek offline, the `volatile` +  `synchronized` definitely make sense to keep implementation thread safe, I am not sure it is worth doing that here: at worst, the `getDockerAvailability()` may be called a few times, but this is no data race and the outcome would be the same (since the environment is the same).

Reply:
@nknize the logic is atomic (the state is a single pointer), the Docker detection logic could run a few times (potentially) which is the only issue I see here, marking `getDockerAvailability()` as `synchronized` would address that (no need for volatile or double check locking) but I have never seen an issue with this service in any builds.